src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -340,5 +349,15 @@ private static void ReturnBlocks(MemoryPoolBlock block, MemoryPoolBlock end)
                 returnBlock.Pool.Return(returnBlock);
             }
         }
+
+        private void ThrowOnConnectionError()
```

Comment:
@davidfowl You're now well-versed in this code. Do you have a better alternative?